js/src/components/paid-ads/ads-campaign/ads-campaign.js

```diff
@@ -0,0 +1,189 @@
+ */
+export default function AdsCampaign( {
+	campaign,
+	headerTitle,
+	onContinue,
+	onSkip = noop,
+	hasError = false,
+	isOnboardingFlow = false,
+	trackingContext,
+} ) {
+	const formContext = useAdaptiveFormContext();
+	const { data: countryCodes } = useTargetAudienceFinalCountryCodes();
+	const { isValidForm, setValue } = formContext;
+	const [ completing, setCompleting ] = useState( null );
+	const [ paidAds, setPaidAds ] = useState( {} );
+
+	useEffect( () => {
+		if ( hasError ) {
+			setCompleting( null );
+		}
+	}, [ hasError ] );
+
+	const handleOnStatesReceived = ( paidAdsValues ) => {
+		setPaidAds( paidAdsValues );
+
+		const { amount } = paidAdsValues;
+		setValue( 'amount', amount );
+	};
+
+	const handleSkipCreateAds = () => {
+		setCompleting( ACTION_SKIP );
+
+		onSkip( paidAds );
+	};
+
+	const handleCompleteClick = ( event ) => {
+		setCompleting( event.target.dataset.action );
+
+		onContinue( paidAds );
+	};
+
+	// The status check of Google Ads account connection is included in `paidAds.isReady`,
+	// because when there is no connected account, it will disable the budget section and set the `amount` to `undefined`.
+	const disabledComplete =
+		completing === ACTION_SKIP || ! paidAds.isReady || ! isValidForm;
+
+	let continueButtonProps = {
+		text: __( 'Continue', 'google-listings-and-ads' ),
+	};
+
+	if ( isOnboardingFlow ) {
+		continueButtonProps = {
+			'data-action': ACTION_COMPLETE,
+			text: __( 'Complete setup', 'google-listings-and-ads' ),
+			eventName: 'gla_onboarding_complete_with_paid_ads_button_click',
+			eventProps: {
+				budget: paidAds.amount,
+				audiences: countryCodes?.join( ',' ),
+			},
+		};
+	}
+
+	let description = createInterpolateElement(
+		__(
+			'Paid Performance Max campaigns are automatically optimized for you by Google. <link>See what your ads will look like.</link>',
+			'google-listings-and-ads'
+		),
+		{
+			link: (
+				<AppDocumentationLink
+					context={ trackingContext }
+					linkId="see-what-ads-look-like"
+					href="https://support.google.com/google-ads/answer/6275294"
+				/>
+			),
+		}
+	);
+
+	if ( isOnboardingFlow ) {
+		description = __(
+			'You’re ready to set up a Performance Max campaign to drive more sales with ads. Your products will be included in the campaign after they’re approved.',
+			'google-listings-and-ads'
+		);
+	}
+
+	return (
+		<StepContent>
+			<StepContentHeader
+				title={ headerTitle }
+				description={ description }
+			/>
+
+			{ isOnboardingFlow && <PaidAdsFeaturesSection /> }
+
+			<PaidAdsSetupSections
+				onStatesReceived={ handleOnStatesReceived }
+				campaign={ campaign }
+				countryCodes={ countryCodes }
+				loadCampaignFromClientSession={ isOnboardingFlow }
+				showCampaignPreviewCard={
+					trackingContext === 'setup-ads' ||
+					trackingContext === 'create-ads'
+				}
```

Comment:
The "same form" means that we will use the same component for both forms so that the requirements of each input section are consistent. The main difference is that the `PaidAdsFeaturesSection` should only be shown during onboarding.

Reply:
Thanks for the explanation.

I would suggest rephrasing it a little bit to avoid having “When the user …, they will see …” wording that confuses its goal of just having to share components in development.